# Module 6 - Critical Thinking - Option 1
## David Edwards
### CSC510 - Foundations of Artificial Intelligence
### Colorado State University-Global Campus
### Dr. Isaac K. Gang
### August 29, 2021



For this assignment, we will be creating a Naive Bayes classifier using Scikit-learn.  

First, we need to import relevant functions.  For this case, we are using the dataset on wine that comes with sklearn.

In [31]:
from sklearn.datasets import load_wine
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB

In [33]:
wine = load_wine()

We'll load the wine dataset which is used to categorize these features into some categories.

In [35]:
wine.feature_names

['alcohol',
 'malic_acid',
 'ash',
 'alcalinity_of_ash',
 'magnesium',
 'total_phenols',
 'flavanoids',
 'nonflavanoid_phenols',
 'proanthocyanins',
 'color_intensity',
 'hue',
 'od280/od315_of_diluted_wines',
 'proline']

So, given values for each of these features, what are we looking to categorize these wines into?  Red, Rose and White?

In [37]:
wine.target_names

array(['class_0', 'class_1', 'class_2'], dtype='<U7')

Oh.  That's not very helpful.  According to [the UCI Machine Learning Repository](https://archive.ics.uci.edu/ml/datasets/wine) these represent wines produced from three different cultivars of grape.  A cultivar represents a plant variety produced by selective breeding. 

Despite looking into the description of the dataset, I can't find the actual cultivar names. 

In [38]:
wine.data.shape

(178, 13)

We have 178 records, each with the 13 feature names shown above

In [42]:
wine.data[0:2]

array([[1.423e+01, 1.710e+00, 2.430e+00, 1.560e+01, 1.270e+02, 2.800e+00,
        3.060e+00, 2.800e-01, 2.290e+00, 5.640e+00, 1.040e+00, 3.920e+00,
        1.065e+03],
       [1.320e+01, 1.780e+00, 2.140e+00, 1.120e+01, 1.000e+02, 2.650e+00,
        2.760e+00, 2.600e-01, 1.280e+00, 4.380e+00, 1.050e+00, 3.400e+00,
        1.050e+03]])

Looking at the data shows the values for each of these features.  I only display the first two rows because the numbers don't mean much to me.

In [8]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=0)

In [20]:
wine = load_wine()

(178, 13)

In [9]:
gnb = GaussianNB()

In [25]:
gnb.fit(X_train, y_train)
y_pred = gnb.predict(X_test)
y_pred
y_prob = gnb.predict_log_proba(X_test)
y_prob

array([[-1.35013778e-10, -2.27256456e+01, -5.97980920e+01],
       [-5.11899676e+01, -5.64985747e+01,  0.00000000e+00],
       [-3.33505966e+00, -3.62620543e-02, -6.62909269e+01],
       [-7.70583597e-12, -2.55889540e+01, -6.71261729e+01],
       [-1.32352445e+01, -1.78651607e-06, -2.79861062e+01],
       [-1.96603825e+00, -1.50835035e-01, -1.09418057e+02],
       [-1.67865721e-12, -2.71124615e+01, -7.09806695e+01],
       [-5.23647854e+01, -2.81848753e+01, -5.75539616e-13],
       [-2.08193504e+01, -9.08395137e-10, -3.23993725e+01],
       [-3.29621299e+01, -5.12715417e-08, -1.67861301e+01],
       [-3.39446337e+01, -1.22459955e+01, -4.80432927e-06],
       [-6.83900737e+01, -2.56645139e+01, -7.14450721e-12],
       [ 0.00000000e+00, -4.80218011e+01, -1.04734637e+02],
       [-2.65688517e-02, -3.64127075e+00, -7.55290075e+01],
       [-3.76830632e+01, -3.82121484e+01,  0.00000000e+00],
       [-2.36963061e+01, -5.11484188e-11, -5.74313673e+01],
       [-2.01637107e-09, -2.00219655e+01

In [11]:
print("Number of mislabeled points out of a total %d points : %d" % (X_test.shape[0], (y_test != y_pred).sum()))

Number of mislabeled points out of a total 89 points : 5
